enhancements/console/pinned-resources.md

```diff
@@ -0,0 +1,166 @@
+
+# Define pre-pinned resources
+
+## Summary
+
+The Developer perspective on the OpenShift Console provides pinned resources on the navigation which enables the user to go to the resource’s list page with a single click.
+
+Some customers want to provide defaults for those pre-pinned resources for all users.
+
+On the Developer perspective navigation, users can see the default pinned resources if they have not customized these pinned resources. The customers have no way to set these default pinned resources for their users.
+
+This enhancement allows cluster admins to define pre-pinned resources for new users and users who have not customized their navigation items.
+
+![Default pinned resources](default-pinned-resources.png)
+
+## Motivation
+
+Some customers want to provide default pre-pinned resources for all users.
+
+### User Stories
+
+#### Story 1
+
+Customer wants to customize the developer perspective navigation by providing pre-pinned resources to users who do not have their navigation items customized.
+
+As a cluster admin, I want to provide default pre-pinned resources for all users.
+
+### Goals
+
+- Allow cluster admins to define default pre-pinned resources on the dev perspective navigation
+
+### Non-Goals
+
+N/A
+
+## Proposal
+
+We can extend the existing `operator.openshift.io/v1` / `Console CRD`.
+We can add `spec.customization.perspectives.pinnedResources` which lists default pre-pinned resources
+
+### Workflow Description
+
+To add default pre-pinned resources:
+
+```text
+├── spec
+│   ├── customization
+|       ├── perspectives
+|           ├── id
+|           ├── ...
+|           ├── pinnedResources
+|                   └──
+└── ...
+```
+
+Each perspective conforms to the following schema :
+
+If the `pinnedResources` list is empty, the default pinned resources will be shown.
+If the `pinnedResources` list is non-empty, only the listed items will be shown if the user hasn't already customized the navigation items.
+
+### API Extensions
+
+Extend the existing `operator.openshift.io/v1` / `Console CRD` by adding `spec.customization.perspectives.pinnedResources`
+
+So the example yaml should look like this:
+
+To set default pre-pinned resources
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Console
+metadata:
+  name: cluster
+  ...
+spec:
+  customization:
+    perspectives:
+    - id: 'dev'
+      ...
+      pinnedResources: ['deployments', 'deploymentconfigs']
+    - ...
+  ...
+```
+
+### Risks and Mitigations
+
+N/A
+
+### Drawbacks
+
+N/A
+
+## Design Details
+
+We will provide a configuration UI and a yaml snippet in the
+console to provide a list of resources that can be pinned to the navigation.
+
+### Test Plan
+
+- If no pinnedResources are defined, then default pinned resources should be visible to users who do not have their navigation items customized
+- If pinnedResources is an empty list, then default pinned resources should be visible to users who do not have their navigation items customized
```

Comment:
where will these default pinned resources be fetched from if pinnedResources is an empty list?

Reply:
they are contributed by the plugins.
Ex: our dev perspective plugin has the default pins
```
{
    "type": "console.perspective",
    "properties": {
      "id": "dev",
      "name": "%devconsole~Developer%",
      "icon": { "$codeRef": "perspective.icon" },
      "defaultPins": [
        {
          "group": "",
          "version": "v1",
          "kind": "ConfigMap"
        },
        {
          "group": "",
          "version": "v1",
          "kind": "Secret"
        }
      ],
      "landingPageURL": { "$codeRef": "perspective.getLandingPageURL" },
      "importRedirectURL": { "$codeRef": "perspective.getImportRedirectURL" },
      "usePerspectiveDetection": { "$codeRef": "perspective.usePerspectiveDetection" }
    }
  },

Reply:
:warning: @debsmita1 @JoelSpeed As this is not our daily business: We CAN (and is it okay??) to differentiate here between not configured (null) and an empty list?